library/network/src/modules/SuSEFirewall.rb

```diff
@@ -1682,6 +1904,112 @@ def GetAdditionalServices(protocol, zone)
       deep_copy(all_allowed_services)
     end
 
+    # Function sets list of services as allowed ports for zone and protocol
```

Comment:
Compared to the original implementation, `SuSEFirewall2::SetAllowedServicesForZoneProto` which [simply sets one backend variable](https://github.com/yast/yast-yast2/blob/8eb9214265fdf8afcae1531c5a0309799028d79d/library/network/src/modules/SuSEFirewall.rb#L533-537), this one is **much** more complex. Can you please add a couple of lines of comments explaining why?
Or even better, break up the 100 line method into 5 methods of 20 lines and the new method names will be explanatory enough.